src/renderer/reader/components/Reader.tsx

```diff
@@ -605,7 +605,11 @@ class Reader extends React.Component<IProps, IState> {
     }
 
     private async handleReadingLocationChange(loc: LocatorExtended) {
+
+        // Why findBookmark here ? can be removed
         this.findBookmarks();
+
+        // every 250ms on user scrolling. This will be saved on db through redux
```

Comment:
The navigator component *debounces*  on the trailing edge of a fast sequence of interactions / events.